src/utils/data-manager.js

```diff
@@ -566,7 +566,7 @@ export default class DataManager {
 
       let object = result;
       object = groups.reduce((o, colDef) => {
-        const value = current[colDef.field] || byString(current, colDef.field);
+        const value = colDef.render ? colDef.render(current) : (current[colDef.field] || byString(current, colDef.field));
```

Comment:
If there is a render function on the column then execute it to get the group-by value

Reply:
But what happens when render returns and react element?

Reply:
That's a good point, I haven't considered that here. 

At the risk of bloating the library, my first thought is to introduce a 'dataAccessor' on the column which could then be used to return primitive values.

What are your thoughts on the best way to handle this scenario?

Reply:
1. field must be filled if we want to group data. 
2. Grouping will work on data[field]
3. render works properly when without grouping.
4. render won't work when column grouped. Because render requires all rowData to work. But we are on a group, not row. We may call render with data[field] instead of all row data. So user must handle this case.

Reply:
I understand each of those points but I'm not sure what you're trying to say there. Can you elaborate?

Reply:
I mean important point is calling render method for a grouped row. First of all, we should find a good solution for it.

Reply:
OK, so at the moment there is a 'type' of group or row which is passed in to the render which allows you to deal with the difference in data type.

Perhaps it would be a neater solution to split it into two props, renderRow and renderGroup. 
That way we can specify through docs and typescript that renderRow may return a component but renderGroup must return a primitive, in order for the groupData() function to work.

Does that sound good to you?

Reply:
No. field already did this. We can call render method with an additional parameter ('row' | 'group') to make a difference in render method according to group or row render.

Reply:
Yeah, absolutely. 

The only issue I see is type safety to ensure that people know they have to be returning a primitive when it is a group.

Is that something that you are happy to leave as just documentation?

Reply:
Yes you can add a blockquote to grouping page on documentation. 

Do you know new documentation is on another repo (github.com/mbrn/material-table.com)

Reply:
Ok, great. Let me make the changes here and if it looks good to you then I'll update the docs to reflect it.